src/Core/src/Platform/iOS/WindowViewController.cs

```diff
@@ -0,0 +1,139 @@
+			_contentWrapperView.AddSubview(windowContentViewController.View);
+		}
+
+		SetUpTitleBar(window, mauiContext, true);
+		AddChildViewController(windowContentViewController);
+		windowContentViewController.DidMoveToParentViewController(this);
+	}
+
+	public override void ViewWillLayoutSubviews()
+	{
+		base.ViewWillLayoutSubviews();
+		LayoutTitleBar();
+	}
+
+	/// <summary>
+	/// Sets up the TitleBar in the ViewController.
+	/// </summary>
+	/// <param name="window">An instance of the <see cref="IWindow"/>.</param>
+	/// <param name="mauiContext">An instance of the <see cref="IMauiContext"/>.</param>
+	/// <param name="isInitalizing"></param>
+	public void SetUpTitleBar(IWindow window, IMauiContext mauiContext, bool isInitalizing)
+	{
+		var platformWindow = window.Handler?.PlatformView as UIWindow;
+
+		if (platformWindow is null || View is null)
+		{
+			return;
+		}
+
+		var newTitleBar = window.TitleBar?.ToPlatform(mauiContext);
+
+		IView? iTitleBar = null;
+		_iTitleBarRef?.TryGetTarget(out iTitleBar);
+
+		if (newTitleBar != iTitleBar)
+		{
+			_titleBar?.RemoveFromSuperview();
+			iTitleBar = null;
+
+			if (newTitleBar is not null)
+			{
+				iTitleBar = window.TitleBar;
+				View.AddSubview(newTitleBar);
+			}
+
+			_titleBar = newTitleBar;
+			_iTitleBarRef = new WeakReference<IView?>(iTitleBar);
+		}
+
+		var platformTitleBar = platformWindow?.WindowScene?.Titlebar;
+
+		if (newTitleBar is not null && platformTitleBar is not null)
+		{
+			platformTitleBar.Toolbar = null;
+			platformTitleBar.TitleVisibility = UITitlebarTitleVisibility.Hidden;
+		}
+
+		LayoutTitleBar();
+
+		// When we are initializing, calling LayoutIfNeeded will cause the layout events to not fire properly.
+		// However we need this when the titlebar is added or removed or the titlebar may not be fully laid out.
+		if (!isInitalizing)
+		{
+			View.LayoutIfNeeded();
+		}
+	}
+
+	/// <summary>
+	/// Measures and arranges the TitleBar and adjusts the frame for the window content to make space for the TitleBar.
+	/// </summary>
+	public void LayoutTitleBar()
+	{
+		_iTitleBarRef.TryGetTarget(out var iTitleBar);
+
+		if (iTitleBar is not null && View is not null)
+		{
+			var measured = iTitleBar.Measure(View.Bounds.Width, double.PositiveInfinity);
+			iTitleBar.Arrange(new Graphics.Rect(0, 0, View.Bounds.Width, measured.Height));
+		}
+
+		var titleBarHeight = iTitleBar?.Frame.Height ?? 0;
+
+		if (!_isTitleBarVisible)
+		{
+			titleBarHeight = 0;
+		}
+
+		var newFrame = new CGRect(0, titleBarHeight, View!.Bounds.Width, View!.Bounds.Height - titleBarHeight);
+
+		if (_contentWrapperView is not null && newFrame != _contentWrapperView.Frame && _contentWrapperView.Subviews.Length > 0)
+		{
+			_contentWrapperView.Frame = newFrame;
+			_contentWrapperView.Subviews[0].Frame = new CGRect(0, 0, View!.Bounds.Width, View!.Bounds.Height - titleBarHeight);
+		}
+	}
+
+	public void SetTitleBarVisibility(bool isVisible) =>
+			_isTitleBarVisible = isVisible;
```

Comment:
Changing the visibility just set the height to zero, right?

Reply:
Yes that is correct!